src/vaadin-message-input.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @license
+ * Copyright (c) 2021 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { PolymerElement, html } from '@polymer/polymer/polymer-element.js';
+import { ThemableMixin } from '@vaadin/vaadin-themable-mixin/vaadin-themable-mixin.js';
+import { ElementMixin } from '@vaadin/vaadin-element-mixin/vaadin-element-mixin.js';
+import '@vaadin/vaadin-button/src/vaadin-button.js';
+import '@vaadin/vaadin-text-field/src/vaadin-text-area.js';
+/**
+ * `<vaadin-message-input>` is a Web Component for sending messages.
+ * It consists of a text area that grows on along with the content, and a send button to send message.
+ *
+ * ```html
+ * <vaadin-message-input></vaadin-message-input>
+ * ```
+ *
+ * ### Styling
+ *
+ * The following shadow DOM parts are available for styling:
+ *
+ * Part name | Description
+ * ----------|----------------
+ *
+ * See [ThemableMixin – how to apply styles for shadow parts](https://github.com/vaadin/vaadin-themable-mixin/wiki)
+ *
+ * @extends HTMLElement
+ * @mixes ThemableMixin
+ * @mixes ElementMixin
+ */
+class MessageInputElement extends ElementMixin(ThemableMixin(PolymerElement)) {
+  static get template() {
+    return html`
+      <style>
+        :host {
+          align-items: flex-start;
+          box-sizing: border-box;
+          display: flex;
+          max-height: 50vh;
+          overflow: hidden;
+        }
+        vaadin-text-area {
+          align-self: stretch;
+          flex-grow: 1;
+          padding: 0;
+        }
+        vaadin-button {
+          flex-shrink: 0;
+          margin: 0;
+        }
+      </style>
+      <vaadin-text-area placeholder="Message"></vaadin-text-area>
+      <vaadin-button theme="primary contained">Send</vaadin-button>
```

Comment:
What's the `contained` theme variant? Can't see it in [the component demo](https://cdn.vaadin.com/vaadin-button/2.4.0/demo/#button-theme-variants-demos).

Reply:
Material theme

Reply:
Apparently this: https://github.com/vaadin/vaadin-button/blob/cc43a29f873186207dafe50217cdf4f7df7d5bac/theme/material/vaadin-button-styles.js#L98